.travis.yml

```diff
@@ -2,29 +2,57 @@ language: rust
 rust: nightly-2019-05-09
 cache: cargo
 
+before_script:
+- >
+  [[ "$(cargo-sweep --version)" == "cargo-sweep 0.4.1" ]]
+  || cargo install cargo-sweep
+- cargo sweep --stamp
+
+before_cache:
+- cargo sweep --file
+
 matrix:
   include:
   - name: cargo doc
     env: [CACHE_NAME=docs]
     script:
-    - RUSTDOCFLAGS=-Dwarnings
-      cargo doc --all --all-features --no-deps --exclude tide
+    - RUSTDOCFLAGS=-Dwarnings cargo doc
+        -Zmtime-on-use
+        --all --all-features
+        --exclude tide
+        --no-deps
 
   - name: cargo fmt
     cache: false
-    before_script: rustup component add rustfmt
-    script: cargo fmt --all -- --check
+    before_script: []
+    install:
+    - rustup component add rustfmt
+    script:
+    - cargo fmt --all -- --check
 
   - name: cargo clippy
     env: [CACHE_NAME=clippy]
-    before_script: rustup component add clippy
-    script: cargo clippy --all --all-targets --exclude examples -- -Dwarnings
+    install:
+    - rustup component add clippy
+    script:
+    - cargo clippy
+        -Zmtime-on-use
+        --all --all-targets
+        --exclude examples
+        -- -Dwarnings
 
   - name: cargo build --no-default-features
     env: [CACHE_NAME=no-default-features]
     script:
-    - cargo build --manifest-path tide/Cargo.toml --no-default-features
-    - cargo build --manifest-path tide-core/Cargo.toml --no-default-features
+    - cargo build
+        -Zmtime-on-use
+        --manifest-path tide-core/Cargo.toml
+        --no-default-features
+    - cargo build
+        -Zmtime-on-use
+        --manifest-path tide/Cargo.toml
+        --no-default-features
```

Comment:
https://github.com/rust-lang/cargo/issues/6972 means that this does some extra work, but it's still faster than the `test` build so doesn't really matter at this point.